src/main/resources/database/changes/release-11/offset_to_static_action_rule_dates.sql

```diff
@@ -0,0 +1,9 @@
+ALTER TABLE action.actionrule
+ADD triggerdatetime timestamptz;
+
+UPDATE action.actionrule
+   SET triggerdatetime = action.case.actionplanstartdate + CONCAT(action.actionrule.daysoffset, ' days')::INTERVAL
+   FROM action.case WHERE action.actionrule.actionplanfk = action.case.actionplanfk;
```

Comment:
Will there definitely be the cases you require in the case table for this to work? Presumably it's possible that there would be action rules which don't currently have any cases in the table

Also couldn't multiple cases satisfy this `WHERE` condition? How does `SET` handle this in that case

Reply:
We have investigated this and there is a risk that cases may not exist for action rules. The triggerdatetime will be set to null if no cases exist, this will need to be manually provisioned as these dates will not be in scope until cases are created.

We checked the query with multiple cases satisfying the where clause and it gets handled without errors as all the cases have the same actionplanstartdate.

Reply:
:+1: Since these dates come from collection exercise we could potentially do a migration to get that data in if it is an issue

Reply:
No need just need to hit the API to update the action rule dates for collection exercises that haven't been executed.